Plugins/Public/base_plugin/BuildModule.cpp

```diff
@@ -171,48 +185,54 @@ bool BuildModule::Timer(uint time)
 	return false;
 }
 
-void BuildModule::LoadState(INI_Reader &ini)
+void BuildModule::LoadState(INI_Reader& ini)
 {
 	while (ini.read_value())
 	{
 		if (ini.is_value("build_type"))
 		{
-			build_type = ini.get_value_int(0);
-		}
-		else if (ini.is_value("paused"))
-		{
-			Paused = ini.get_value_bool(0);
-		}
-		else if (ini.is_value("produced_item"))
-		{
-			active_recipe.produced_item = ini.get_value_int(0);
-		}
-		else if (ini.is_value("cooking_rate"))
-		{
-			active_recipe.cooking_rate = ini.get_value_int(0);
+			active_recipe = moduleNumberRecipeMap[ini.get_value_int(0)];
```

Comment:
In case of a std::map or std::unordered_map, attempt to read a nonexistent element from a map simply calls a default constructor. I suppose I could still add some checks but what this code would result in would be simply loading an empty recipe that would instantly complete and disappear without adding anything to the base, effectively silently quitting.